### PR TITLE
WIP: re-arrange folders and imports

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

**Comment by jeffbcross** [comment]
This PR is getting closer to being green and ready to review. There are still some TODOs on #3713 to do in follow-up commits, and then #3714 should follow soon-after.

File: modules/angular2/src/core/di/index.ts [hunk 1]
```diff
@@ -1,8 +1,4 @@
-/**
- * @module
- * @description
- * The `di` module provides dependency injection container services.
- */
+// Public exports for distribution as standalone package
```
Comment (jeffbcross) [inline modules/angular2/src/core/di/index.ts]: This file is not currently being used, but should be the default export when DI is published as an npm package.

**Comment by jeffbcross** [comment]
I should squash some commits, like multiple commits of moving around directives.

**Comment by jeffbcross** [comment]
@jteplitz602 would you mind taking a look at the changes to `web_worker` imports?

@mhevery this should be good to review now. If there are Travis failures at this point, they are likely not significant.

**Comment by jeffbcross** [comment]
I'm troubleshooting the issue with E2E tests failing. Some SystemJS module dependency race condition.

File: modules/angular2/web_worker/worker.ts [hunk 2]
```diff
@@ -1,9 +1,5 @@
 export * from '../metadata';
 export * from '../change_detection';
 export * from '../core';
```
Comment (jteplitz602) [inline modules/angular2/web_worker/worker.ts]: I think this should be core_no_render

**Comment by jeffbcross** [comment]
I'm going to cleanup and fix some things, won't review this yet.

**Comment by jeffbcross** [comment]
Closing PR to re-work and re-open

**Pull request closed without merging**